src/rust/engine/src/externs/target.rs

```diff
@@ -188,32 +230,58 @@ impl Field {
             _ => Ok(py.NotImplemented()),
         }
     }
+
+    fn __getattribute__<'a>(
+        self_: &'a PyCell<Self>,
+        name: String,
+        py: Python<'a>,
+    ) -> PyResult<*mut ffi::PyObject> {
+        if name == "default" {
+            if let Some(default) = &self_.extract::<Self>()?.default {
+                // Return instance default, overriding the field class var default.
+                return Ok(default.as_ptr());
+            }
+        }
+
+        unsafe {
+            // The ffi::PyObject_GenericGetAttr() call is unsafe, so we need to be in an unsafe
+            // context to call it.
+            let slf = self_.borrow_mut().into_ptr();
+            let attr = name.into_py(py).into_ptr();
+            let res = ffi::PyObject_GenericGetAttr(slf, attr);
+            if res.is_null() {
+                Err(PyErr::fetch(py))
+            } else {
+                Ok(res)
+            }
+        }
+    }
```

Comment:
there ought to be a better way to do this...
https://stackoverflow.com/questions/78522334/how-do-i-call-the-equiv-of-object-getattribute-self-name-using-pyo3